src/main/java/org/saltyrtc/client/signaling/Signaling.java

```diff
@@ -315,9 +324,23 @@ public void onConnected(WebSocket websocket, Map<String, List<String>> headers)
             @SuppressWarnings("UnqualifiedMethodAccess")
             public void onConnectError(WebSocket websocket, WebSocketException ex) throws Exception {
                 getLogger().error("Could not connect to websocket (" + ex.getError().toString() + "): " + ex.getMessage());
-                if (Signaling.this.wsConnectAttempt == 1) {
-                    getLogger().info("Retrying to reconnect once...");
+                if (Signaling.this.wsConnectAttemptsMax <= 0 || Signaling.this.wsConnectAttempt < Signaling.this.wsConnectAttemptsMax) {
+                    // Increase #attempts (and timeout if needed)
+                    if (Signaling.this.wsConnectLinearBackoff) {
+                        Signaling.this.wsConnectTimeout += Signaling.this.wsConnectTimeoutInitial;
+                    }
                     Signaling.this.wsConnectAttempt += 1;
+
+                    // Log retry attempt
+                    String retryConstraint;
```

Comment:
This *could* be made `final` too, but feel free to leave it as-is. Just a personal perference for immutable variable bindings everywhere :)